src/Engine/Graphics/Collisions.cpp

```diff
@@ -749,129 +872,147 @@ void ProcessPartyCollisionsBLV(int sectorId, int min_party_move_delta_sqr, int *
+            // TODO(pskelton): common to odm/blv so extract
+            Vec3f newdirection;
+            if (collision_state.adjusted_move_distance > 0.0f) {
+                // Create new sliding plane from collisio
+                Vec3f slideplaneorigin = collision_state.collisionPos;
+                Vec3f slideplanenormal = collision_state.collisionNorm;
+                float slideplanedist = -(dot(slideplaneorigin, slideplanenormal));
+
+                // Form a sliding vector that is parallel to sliding movement
+                // Take where you wouldve ended up without collisions and move that onto the slide plane by adding the normal
+                // Start point to new destination is a vector along the slide plane
+                float destplanedist = dot(collision_state.new_position_lo, slideplanenormal) + slideplanedist;
+                Vec3f newdestination = collision_state.new_position_lo - destplanedist * slideplanenormal;
+                newdirection = newdestination - collision_state.collisionPos;
+                newdirection.z = 0;
+                newdirection.normalize();
+            }
+
+            // Set party to move along this new sliding vector
+            pParty->speed = newdirection * dot(newdirection, pParty->speed);
+            // Skip reducing party speed
+            continue;
         }
 
         if (collision_state.pid.type() == OBJECT_Face) {
             BLVFace *pFace = &pIndoor->pFaces[collision_state.pid.id()];
+            bool bFaceSlopeTooSteep = pFace->facePlane.normal.z > 0.0f && pFace->facePlane.normal.z < 0.70767211914f; // Was 46378 fixpoint
+
+            // TODO(pskelton): Better way to do this?
+            // Special case for steep staircase in tidewater
+            if (pCurrentMapName == "D17.blv") {
+                if (collision_state.pid.id() == 650)
+                    bFaceSlopeTooSteep = false;
+            }
+
+            // new sliding plane
+            Vec3f slideplaneorigin = collision_state.collisionPos;
+            Vec3f slideplanenormal = adjusted_pos + Vec3f(0, 0, collision_state.radius_lo) - slideplaneorigin;
+            slideplanenormal.normalize();
+            float slideplanedist = -(dot(slideplaneorigin, slideplanenormal));
+            float distfromdestpointtoplane = dot(collision_state.new_position_lo, slideplanenormal) + slideplanedist;
+            Vec3f newdestination = collision_state.new_position_lo - distfromdestpointtoplane * slideplanenormal;
+            Vec3f newdirection = newdestination - collision_state.collisionPos;
+
+            // Cant push uphill on steep faces
+            if (bFaceSlopeTooSteep && newdirection.z > 0)
+                newdirection.z = 0;
+
+            newdirection.normalize();
+
+            // Push away from the surface and add a touch down for better slide
+            if (bFaceSlopeTooSteep)
+                pParty->speed += Vec3f(pFace->facePlane.normal.x, pFace->facePlane.normal.y, -2) * 10;
```

Comment:
Is this needed? Do we end up inside polygons w/o this?

Reply:
Because of the way the new sliding velocity is calculated, if party is standing on a surface the offset is small and resulting vector is small. This gives a better downhill push and smoother slide.